### PR TITLE
feat: Add Volo.Abp.Quartz.Database Module

### PR DESCRIPTION
**Pull request by JadynWong**
Related #3433

Using EfCore to generate Quartz databases。

I tested it on sqlserver,mysql,postgresql

If this module is accepted,maybe these host projects should move to abp-samples project?

### CODE REVIEW

File: modules/quartz-database/src/Volo.Abp.Quartz.Database.EntityFrameworkCore/Volo/Abp/Quartz/Database/EntityFrameworkCore/QuartzDatabaseDbContextModelCreatingExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+{
+    public static class QuartzDatabaseDbContextModelCreatingExtensions
+    {
+        public static void ConfigureQuartzDatabase(
+            this ModelBuilder builder,
+            Action<QuartzDatabaseModelBuilderConfigurationOptions> optionsAction = null)
+        {
+            Check.NotNull(builder, nameof(builder));
+
+            //TODO: sqlite database need to create triggers, any way?
```
Comment (realLiangshiwei) [inline modules/quartz-database/src/Volo.Abp.Quartz.Database.EntityFrameworkCore/Volo/Abp/Quartz/Database/EntityFrameworkCore/QuartzDatabaseDbContextModelCreatingExtensions.cs:16]: Can you explain in detail?

Comment (JadynWong) [reply modules/quartz-database/src/Volo.Abp.Quartz.Database.EntityFrameworkCore/Volo/Abp/Quartz/Database/EntityFrameworkCore/QuartzDatabaseDbContextModelCreatingExtensions.cs:16]: https://github.com/quartznet/quartznet/blob/41846f43abfcfa45c2761a7fb93ac376a6b65d07/database/tables/tables_sqlite.sql#L65-L68

SQLite uses database triggers for cascading delete.  I did not find a way to automatically use efcore to generate these triggers.

[https://www.sqlite.org/foreignkeys.html](https://www.sqlite.org/foreignkeys.html)

Maybe the newer version of sqlite supports `ON DELETE` action, I'm not sure.

**Comment by realLiangshiwei** [comment]
See https://github.com/abpframework/abp/issues/3433#issuecomment-887943917

File: modules/quartz-database/src/Volo.Abp.Quartz.Database.EntityFrameworkCore/Volo/Abp/Quartz/Database/EntityFrameworkCore/QuartzDatabaseDbContextModelCreatingExtensions.cs (hunk 1 above)
Comment (JadynWong) [reply modules/quartz-database/src/Volo.Abp.Quartz.Database.EntityFrameworkCore/Volo/Abp/Quartz/Database/EntityFrameworkCore/QuartzDatabaseDbContextModelCreatingExtensions.cs:16]: Although this PR is closed, I would like to state the results about this comment.

The sqlite connection string uses "Foreign Keys=true" to enable "foreign key". For example: `Data Source=QuartzDatabaseDemo.db;Foreign Keys=true`.

[https://docs.microsoft.com/en-us/dotnet/standard/data/sqlite/connection-strings#foreign-keys](https://docs.microsoft.com/en-us/dotnet/standard/data/sqlite/connection-strings#foreign-keys)

**Pull request closed without merging**